File: src/main/java/com/todolist/task/service/domain/TaskRepository.java
package com.todolist.task.service.domain;

import com.todolist.task.service.domain.models.Task;
import java.util.List;

public interface TaskRepository {

  void save(Task task);

  List<Task> findByCreatedByAndDeletedFalse(String user);

  Task findById(String taskId);
}

File: src/main/kotlin/com/todolist/task/service/domain/TaskRepository.kt
package com.todolist.task.service.domain

import com.todolist.task.service.domain.models.Task

interface TaskRepository {
    fun save(task: Task)
    fun findByCreatedByAndDeletedFalse(user: String): List<Task>?
    fun findById(taskId: String): Task?
}
